sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -0,0 +1,389 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import fs from "fs-extra";
+import nise from "nise";
+import queryString from "query-string";
+import {
+  isBrowser,
+  blobToString,
+  escapeRegExp,
+  env,
+  TestInfo,
+  isPlayingBack,
+  isRecording
+} from "./utils";
+import { customConsoleLog } from "./customConsoleLog";
+
+let nock: any;
+
+let replaceableVariables: { [x: string]: string } = {};
+export function setReplaceableVariables(a: { [x: string]: string }): void {
+  replaceableVariables = a;
+  if (isPlayingBack()) {
+    // Providing dummy values to avoid the error
+    Object.keys(a).map((k) => {
+      env[k] = a[k];
+    });
+  }
+}
+
+let replacements: any[] = [];
+export function setReplacements(maps: any): void {
+  replacements = maps;
+}
+
+export function setEnviromentOnLoad() {
+  if (!isBrowser() && (isRecording || isPlayingBack)) {
+    nock = require("nock");
+  }
+
+  if (isBrowser() && isRecording()) {
+    customConsoleLog();
+  }
+
+  if (isPlayingBack()) {
+    // Providing dummy values to avoid the error [ENVs for storage packages]
+    env.ACCOUNT_NAME = "fakestorageaccount";
+    env.ACCOUNT_KEY = "aaaaa";
+    env.ACCOUNT_SAS = "aaaaa";
+    env.STORAGE_CONNECTION_STRING = `DefaultEndpointsProtocol=https;AccountName=${
+      env.ACCOUNT_NAME
+    };AccountKey=${env.ACCOUNT_KEY};EndpointSuffix=core.windows.net`;
+  }
+}
+
+const skip = [
+  // Abort
+  "browsers/aborter/recording_should_abort_after_aborter_timeout.json"
+];
+
+export abstract class BaseRecorder {
+  protected readonly filepath: string;
+  public uniqueTestInfo: TestInfo = { uniqueName: {}, newDate: {} };
+
+  constructor(runtime: string, testHierarchy: string, testTitle: string, ext: string) {
+    this.filepath =
+      runtime +
+      "/" +
+      this.formatPath(testHierarchy) +
+      "/recording_" +
+      this.formatPath(testTitle) +
+      "." +
+      ext;
+  }
+
+  protected formatPath(path: string): string {
+    return path
+      .toLowerCase()
+      .replace(/ /g, "_")
+      .replace(/<=/g, "lte")
+      .replace(/>=/g, "gte")
+      .replace(/</g, "lt")
+      .replace(/>/g, "gt")
+      .replace(/=/g, "eq")
+      .replace(/\W/g, "");
+  }
+
+  /**
+   * Additional layer of security to avoid unintended/accidental occurrences of secrets in the recordings
+   * */
+  protected filterSecrets(recording: string): string {
+    let updatedRecording = recording;
+    for (const k of Object.keys(replaceableVariables)) {
+      const escaped = escapeRegExp(env[k]);
+      updatedRecording = updatedRecording.replace(
+        new RegExp(escaped, "g"),
+        replaceableVariables[k]
+      );
+    }
+    for (const map of replacements) {
+      updatedRecording = map(updatedRecording);
+    }
+
+    // Handling storage environment variables separately
```

Comment:
Same as https://github.com/Azure/azure-sdk-for-js/pull/4281/files?file-filters%5B%5D=.json&file-filters%5B%5D=.ts#r308844665, add a TODO comment

Reply:
Added